src/clj/puppetlabs/services/ca/certificate_authority_service.clj

```diff
@@ -55,23 +55,24 @@
                        events)
              (ca/retrieve-ca-crl! ca-crl-file host-crl-file)))))
       (assoc context :auth-handler auth-handler
-                     :watcher watcher)))
+                     :watcher watcher
+                     :ca-settings settings)))
 
   (initialize-master-ssl!
-   [this master-settings certname]
-   (let [settings (ca/config->ca-settings (get-config))]
-     (ca/initialize-master-ssl! master-settings certname settings)))
+     [this master-settings certname]
+     (let [settings (tk-services/service-context this)]
+       (ca/initialize-master-ssl! master-settings certname (:ca-settings settings))))
 
   (retrieve-ca-cert!
-    [this localcacert]
-    (ca/retrieve-ca-cert! (get-in-config [:puppetserver :cacert])
-                          localcacert))
+     [this localcacert]
```

Comment:
@justinstoller and now I think I've added too much whitespace... Should that space before ``[this localcacert]`` and the next line be removed?

Reply:
yes